flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGateMetrics.java

```diff
@@ -86,6 +86,9 @@ int refreshAndGetMin() {
 			}
 		}
 
+		if (min == Integer.MAX_VALUE) { // in case all channels are local
```

Comment:
Is it worth removing the earlier check for `channels.isEmpty()` and merging the comments?